### PR TITLE
x86 fixes part 2

### PR DESCRIPTION
**Pull request by LarryOsterman**
# Fixed EventHubs nightly build problems.

This fixes a number of build breaks in the eventhubs tests which show up with the Visual Studio 2019 msbuild CMAKE generator on x86 builds only.

## Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

**Comment by LarryOsterman** [comment]
azp run cpp - core

File: CMakePresets.json [hunk 1]
```diff
@@ -57,6 +78,18 @@
...
+      "name": "x86-msvc-static",
+      "displayName": "Windows x86 MSVC Static",
+      "description": "Windows Default, MSVC, x86 architecture.",
+      "inherits": "msvc-windows-default",
+      "hidden": true,
+      "cacheVariables": {
+        "VCPKG_TARGET_TRIPLET": "x86-windows-static",
+        "MSVC_USE_STATIC_CRT": true
+      },
+      "architecture": "win32"
```
Comment (ahsonkhan) [inline CMakePresets.json:91]: Should we add this `architecture` property to the other x86-* presets being added below?

File: sdk/eventhubs/azure-messaging-eventhubs/README.md [hunk 2]
```diff
@@ -1,3 +1,3 @@
 # Azure Messaging EventHub service
 
-C++ SDK for Azure Messaging EventHub.
\ No newline at end of file
+C++ SDK for Azure Messaging EventHub.
```
[the remark this answers is not in the document]
Comment (LarryOsterman) [reply sdk/eventhubs/azure-messaging-eventhubs/README.md]: THere's a crlf at the end of the file in my version, none on the original.

File: CMakePresets.json (hunk 1 above)
Comment (LarryOsterman) [reply CMakePresets.json:91]: All the x86 presets being added should derive from the MSVC base and the way cmake presets work, derived presets include all the settings from their parent.

File: CMakePresets.json [hunk 3]
```diff
@@ -289,6 +344,16 @@
...
     },
+    {
+      "name": "x86-msvc-static-debug-perftests",
+      "displayName": "x86 MSVC Debug static With Perf Tests and samples",
+      "inherits": "x86-msvc-static-debug-tests",
+      "cacheVariables": {
+        "BUILD_TRANSPORT_WINHTTP": true,
+        "BUILD_PERFORMANCE_TESTS": true,
+        "BUILD_SAMPLES": true
+      }
```
Comment (ahsonkhan) [inline CMakePresets.json:355]: It isn't clear what the difference is between `x86-msvc-static-debug-perftests` and `x86-msvc-static-debug-tests` from the name/description. It seems the only difference is that building on winhttp is true, in one.

File: CMakePresets.json (hunk 1 above)
Comment (LarryOsterman) [reply CMakePresets.json:91]: It's part of why I love the cmakepresets.json stuff - it would be nice if there was platform matrix support like we have in the CI pipelines, but it's really nice that you get a hierarchy of types.

File: CMakePresets.json (hunk 3 above)
Comment (LarryOsterman) [reply CMakePresets.json:355]: A lot of this was transcribed from our old cmakesettings.json files, which were sloppy. When I created the cmakepresets.json, I didn't want to break anyone's workflow so I copied a bunch of the stuff from cmakesettings.json.

THis is an example of that - -debug-tests is only supposed to enable build_testing and build_samples, the debug-perftests is supposed to enable perf tests on top of -debug-tests. 

I'll straighten this out because it's annoying.

File: CMakePresets.json [hunk 4]
```diff
@@ -88,140 +121,73 @@
...
-      "displayName": "Windows x86 Release, static",
-      "description": "Windows x86 Release, static",
-      "inherits": "x86-static",
-      "hidden": true,
-      "cacheVariables": {
-        "CMAKE_BUILD_TYPE": "RelWithDebInfo"
-      }
+      "name": "x86-msvc-static-debug",
+      "displayName": "Windows x86, msvc static, Debug ",
+      "description": "Windows x86, msvc static, Debug",
```
Comment (ahsonkhan) [inline CMakePresets.json:155]: If hiding this was intentional:
```suggestion
      "description": "Windows x86, msvc static, Debug (hidden)",
```

File: CMakePresets.json [hunk 5]
```diff
@@ -88,140 +121,73 @@
...
-      "name": "x64-static-release",
-      "displayName": "Windows x64 Release, static",
-      "description": "Windows x64 Release, static",
-      "inherits": "x64-static",
-      "hidden": true,
-      "cacheVariables": {
-        "CMAKE_BUILD_TYPE": "RelWithDebInfo"
+        "BUILD_TESTING": true,
+        "ENABLE_PROXY_TESTS": false,
+        "BUILD_SAMPLES": true
```
Comment (ahsonkhan) [inline CMakePresets.json:149]: Do we always want to couple building samples when enabling tests?

For example, we previously weren't building samples for the `x86-static-debug-tests` preset, but now we will, because of
> "inherits": [ "x86-static", "debug-build", "enable-tests" ]

File: CMakePresets.json [hunk 6]
```diff
@@ -88,140 +121,73 @@
...
-      }
-    },
-    {
-      "name": "x86-release",
-      "displayName": "x86 Release",
-      "description": "Windows x86 Release",
-      "inherits": "x86",
+      "name": "enable-tests",
+      "displayName": "Enable Tests",
+      "description": "Enable Tests (hidden)",
```
Comment (ahsonkhan) [inline CMakePresets.json:144]: nit: consistent casing
```suggestion
      "description": "Enable Tests (Hidden)",
```

File: CMakePresets.json [hunk 7]
```diff
@@ -88,140 +121,73 @@
...
+      "displayName": "x64 Debug With Tests",
       "description": "Windows x64 Debug build with Tests configured",
-      "inherits": "x64-debug",
-      "cacheVariables": {
-        "BUILD_TESTING": true,
-        "ENABLE_PROXY_TESTS": false,
-        "BUILD_SAMPLES": true,
-        "BUILD_PERFORMANCE_TESTS": true,
-        "BUILD_TRANSPORT_WINHTTP": true,
-        "BUILD_TRANSPORT_CURL": true
```
Comment (ahsonkhan) [inline CMakePresets.json:205]: This preset is no longer setting winhttp/curl to true. It's a bit hard to follow in a diff'd review, so can you point me to the name of the preset which now has this, since we removed `winhttp+curl` from this one's name?

File: CMakePresets.json (hunk 5 above)
Comment (LarryOsterman) [reply CMakePresets.json:149]: We could absolutely add samples as a separate dimension. 

Different question: When would we want to build tests without samples?

Comment (ahsonkhan) [reply CMakePresets.json:149]: Usually I build tests so I can run them, to validate correctness in my inner dev loop/locally. That's pretty common.

Since we tend not to have breaking changes, samples are typically fairly stable, and I am happy with CI running them to make sure they are written correctly and don't accidentally break. The times where I care to build samples is much lower than tests.

Comment (LarryOsterman) [reply CMakePresets.json:149]: That's reasonable. I'll make that change.

File: CMakePresets.json (hunk 7 above)
Comment (LarryOsterman) [reply CMakePresets.json:205]: x64-static-debug-perftests has both curl and winhttp. It's basically the kitchen sink (similar to x64-debug-tests).

Comment (LarryOsterman) [reply CMakePresets.json:205]: x64-debug-tests is exactly what it says it is. x64, debug, tests (the old name was wrong since it included perf tests, and samples).

**Comment by LarryOsterman** [comment]
Ping. This PR just has fixes for the nightly build and it would be nice to unblock all the C++ builds.

**Review by ahsonkhan — APPROVED** [review]
Lgtm

Splitting it was a great idea!
